pkg/pillar/cmd/baseosmgr/baseosmgr.go

```diff
@@ -434,27 +432,20 @@ func handleGlobalConfigModify(ctxArg interface{}, key string,
 		log.Infof("handleGlobalConfigModify: ignoring %s\n", key)
 		return
 	}
-	var gcp *types.GlobalConfig
-	debug, gcp = agentlog.HandleGlobalConfig(ctx.subGlobalConfig, agentName,
-		debugOverride)
+	var gcp *types.ConfigItemValueMap
+	debug, gcp = agentlog.HandleGlobalConfig(ctx.subGlobalConfig, agentName, debugOverride)
 	if gcp != nil {
-		ctx.globalConfig = gcp
+		ctx.configItemValueMap = gcp
 	}
 	log.Infof("handleGlobalConfigModify done for %s\n", key)
 }
 
 func handleGlobalConfigDelete(ctxArg interface{}, key string,
 	statusArg interface{}) {
-
-	ctx := ctxArg.(*baseOsMgrContext)
 	if key != "global" {
 		log.Infof("handleGlobalConfigDelete: ignoring %s\n", key)
 		return
 	}
-	log.Infof("handleGlobalConfigDelete for %s\n", key)
-	debug, _ = agentlog.HandleGlobalConfig(ctx.subGlobalConfig, agentName,
-		debugOverride)
-	*ctx.globalConfig = types.GlobalConfigDefaults
```

Comment:
Why are you deleting this code?

Reply:
I was told that the delete function is not actually ever run. I used to have 

ctx.configValueItemMap = type.GetDefualtValueMap(ctx.configSpecMap)

but this would require this context to have the specMap, which I was told should only exist in zedagent during init. Is what I had earlier fine?

Reply:
Is this correct? or should I be setting everything to default values?